Source/santad/SNTEndpointSecurityManager.m

```diff
@@ -0,0 +1,159 @@
+/// Copyright 2019 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+#import "Source/santad/SNTEndpointSecurityManager.h"
+
+#import "Source/common/SNTLogging.h"
+
+#include <EndpointSecurity/EndpointSecurity.h>
+
+#include <bsm/libbsm.h>
+
+@interface SNTEndpointSecurityManager ()
+
+@property(nonatomic) es_client_t *client;
+@property (nonatomic, copy) void (^decisionCallback)(santa_message_t);
+@property (nonatomic, copy) void (^logCallback)(santa_message_t);
+
+@end
+
+@implementation SNTEndpointSecurityManager
+
+- (instancetype)init API_AVAILABLE(macos(10.15)) {
+  self = [super init];
+  if (self) {
+    es_client_t *client = NULL;
+    es_new_client_result_t ret = es_new_client(&client, ^(es_client_t *c, const es_message_t *m) {
+      // TODO(bur/rah): Currently this class only subscribes to exec events. Move this code
+      // somewhere exec specific if other types of events are added to this client.
+      santa_message_t sm;
+      sm.uid = audit_token_to_ruid(m->event.exec.target->audit_token);
+      sm.gid = audit_token_to_rgid(m->event.exec.target->audit_token);
+      sm.pid =audit_token_to_pid(m->event.exec.target->audit_token);
+      sm.ppid = m->event.exec.target->ppid;
+      sm.es_message = (void *)es_copy_message(m);
+
+      size_t l = m->event.exec.target->executable->path.length;
+      if (l + 1 > MAXPATHLEN ||  m->event.exec.target->executable->path_truncated) {
+        // TODO(bur/rah): Get path from fsid.
+        LOGE(@"Path is truncated!");
+        es_respond_auth_result(self.client, m, ES_AUTH_RESULT_ALLOW, true);
+        return;
+      }
+      strncpy(sm.path,m->event.exec.target->executable->path.data, l);
+      sm.path[l] = '\0';
+
+      switch (m->event_type) {
+        case ES_EVENT_TYPE_AUTH_EXEC:
+          // TODO(bur/rah): Probably also want to do this for is_es_client.
+          // TODO(bur/rah): Since these events are not evaluated by Santa's pipline they are
+          //                missing bits of information such as SHA256 and REASON. Refactor the
+          //                logging cache.
+          if (m->event.exec.target->is_platform_binary) {
```

Comment:
I don't think we should ignore _all_ platform binaries - doesn't that include everything that's part of the base OS? We have some users who like to block things like iTunes, by hash, etc.

Reply:
Yeah, I naively thought that "is_platform_binary" was just the "CoreOS" bits. That doesn't seem to be the case. /usr/bin/yes for example has "is_platform_binary" marked as true.

Commenting this out for now.